src/components/Dashboard/borrow_markets/borrowMarketsTable.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { useEffect } from "react";
+import Table from "@mui/material/Table";
+import TableBody from "@mui/material/TableBody";
+import TableCell from "@mui/material/TableCell";
+import TableHead from "@mui/material/TableHead";
+import TableRow from "@mui/material/TableRow";
+
+import { Box, Paper, TableContainer, Typography } from "@mui/material";
+
+// Token Logos
+import sxTokenLogo from "../../../assets/img/sx_coin_token.png";
+import usdcTokenLogo from "../../../assets/img/usdc_coin_token.png";
+
+// Action Items
+import { useDispatch, useSelector } from "react-redux";
+import { AppDispatch, RootState } from "../../../app/Store";
+import WSXBorrowMarketDialog from "./wsxBorrowMarketDialog";
+import USDCBorrowMarketDialog from "./usdcBorrowMarketDialog";
+import { updateUSDCBalance, updateUSDCBorrowRate, updateUSDCOraclePrice } from "../../../features/dashboard/USDCMarketSlice";
+import { updateWSXBalance, updateWSXBorrowRate, updateWSXOraclePrice } from "../../../features/dashboard/WSXMarketSlice";
+
+export default function BorrowMarkets() {
+  const [borrowSXDialogOpen, setBorrowSXDialogOpen] = React.useState(false);
+  const [borrowUSDCDialogOpen, setBorrowUSDCDialogOpen] = React.useState(false);
+
+  const dispatch = useDispatch<AppDispatch>();
+
+  const usdcBorrowAPY = useSelector(
+    (state: RootState) => state.usdc.borrowRate
+  );
+  const usdcWalletBalance = useSelector(
+    (state: RootState) => state.usdc.balance
+  );
+  const usdcOraclePrice = useSelector(
+    (state: RootState) => state.usdc.oraclePrice
+  );
+
+  const wsxBorrowAPY = useSelector((state: RootState) => state.wsx.borrowRate);
+  const wsxWalletBalance = useSelector(
+    (state: RootState) => state.wsx.balance
+  );
+  const wsxOraclePrice = useSelector(
+    (state: RootState) => state.wsx.oraclePrice
+  );
+
+  function handleSXRowClick(event: React.MouseEvent) {
+    setBorrowSXDialogOpen(true);
+  }
+
+  function handleUSDCRowClick(event: React.MouseEvent) {
+    setBorrowUSDCDialogOpen(true);
+  }
+
+  useEffect( () => {
+    // update borrow apys, wallet balances, and oracle prices
+    
+    dispatch(updateUSDCBorrowRate());
+    dispatch(updateWSXBorrowRate());
+
+    dispatch(updateWSXBalance());
+    dispatch(updateUSDCBalance());
+    
+    dispatch(updateUSDCOraclePrice());
+    dispatch(updateWSXOraclePrice());
+  });
```

Comment:
yes it is meant to dispatch every render. We need accurate values readable from the blockchain.

Reply:
What is a dependency array?

Reply:
In the context of `useEffect` hook, it is used to control when the effect should run.

```
useEffect(() => {
        countRef.current = count; // Update the ref value
      }, [count]);
```
This for example, the effect will run during the first render and if `count` changes. It is used to prevent unnecessary re-renders.